### PR TITLE
Migrate to Californium 3.2.0.

### PR DESCRIPTION
**Pull request by n-deliyski**
Signed-off-by: Nikolay Deliyski <Nikolay.Deliyski@bosch.io>
Also-by: Achim Kraus <achim.kraus@bosch.io>

### CODE REVIEW

**Comment by boaks** [comment]
@sophokles73 

Just to mention:
[Californium 3.4](https://github.com/eclipse/californium/issues/1952) will be released tomorrow. Anyway, I would prefer to first migrate to 3.2 and afterwards discus/apply an alternative for PR #3176 and the update to Californium 3.4 with that.

File: adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DeviceRegistryBasedCertificateVerifier.java [hunk 1]
```diff
@@ -214,46 +212,42 @@ public void setResultHandler(final HandshakeResultHandler resultHandler) {
...
             final ConnectionId cid,
             final ServerNames serverName,
-            final Boolean clientUsage,
-            final boolean truncateCertificatePath,
-            final CertificateMessage message,
-            final DTLSSession session) {
+            final InetSocketAddress remotePeer,
+            final boolean clientUsage,
+            final boolean verifySubject,
+            final boolean truncateCertificatePath, final CertificateMessage message) {
```
Comment (sophokles73) [inline adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DeviceRegistryBasedCertificateVerifier.java:226]: can you please put the last parameter on its own line as well?
*verifySubject* and *truncateCertificatePath* are not used, or am I mistaken?

File: adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DeviceRegistryBasedCertificateVerifier.java [hunk 2]
```diff
@@ -214,46 +212,42 @@ public void setResultHandler(final HandshakeResultHandler resultHandler) {
...
-            if (certificates.isEmpty()) {
-                final AlertMessage alert = new AlertMessage(
-                        AlertLevel.FATAL,
-                        AlertDescription.BAD_CERTIFICATE,
-                        session.getPeer());
+            final List<? extends Certificate> list = certChain.getCertificates();
+            final int pathSize = list.size();
+            if (pathSize < 1) {
+                final AlertMessage alert = new AlertMessage(AlertLevel.FATAL,
+                        AlertDescription.BAD_CERTIFICATE);
```
Comment (sophokles73) [inline adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DeviceRegistryBasedCertificateVerifier.java:239]: any particular reason to change this code?

File: adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DeviceRegistryBasedPskStore.java [hunk 3]
```diff
@@ -194,14 +194,15 @@ public PskSecretResult requestPskSecretResult(
             final PskPublicInformation identity,
             final String hmacAlgorithm,
             final SecretKey otherSecret,
-            final byte[] seed) {
+            final byte[] seed,
+            final boolean useExtendedMasterSecret) {
```
Comment (sophokles73) [inline adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DeviceRegistryBasedPskStore.java:198]: useExtendedMasterSecret does not seem to be used?

**Review by sophokles73 — CHANGES_REQUESTED** [review]

File: adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DeviceRegistryBasedCertificateVerifier.java (hunk 2 above)
Comment (boaks) [reply adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DeviceRegistryBasedCertificateVerifier.java:239]: In Californium 2.7 the `clientUsage` was `Boolean` and so `if (clientUsage != null) {` is now obsolete. The  other changes may be re-edited towards the previous terms and coding style in hone.

File: adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DeviceRegistryBasedPskStore.java (hunk 3 above)
Comment (boaks) [reply adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DeviceRegistryBasedPskStore.java:198]: Yes, it's not used. But it's part of the [callback API](https://github.com/eclipse/californium/blob/master/scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/pskstore/AdvancedPskStore.java#L61-L62).
Any rules you want to be obeyed for not used parameter in callbacks?

Comment (sophokles73) [reply adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DeviceRegistryBasedPskStore.java:198]: Ah, my bad. I didn't see that this has an `@Override` annotation ...

File: adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DeviceRegistryBasedCertificateVerifier.java (hunk 2 above)
Comment (sophokles73) [reply adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DeviceRegistryBasedCertificateVerifier.java:239]: yes, please do so

Comment (n-deliyski) [reply adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DeviceRegistryBasedCertificateVerifier.java]: the original code style is reverted

File: adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DeviceRegistryBasedCertificateVerifier.java (hunk 1 above)
Comment (n-deliyski) [reply adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DeviceRegistryBasedCertificateVerifier.java]: done

**Pull request merged**